pallets/gear/src/mock.rs

```diff
@@ -315,3 +322,182 @@ pub fn calc_handle_gas_spent(source: H256, dest: H256, payload: Vec<u8>) -> (u64
+    T::GasHandler::create(source.into_origin(), root_message_id, initial_gas)
+        .map_err(|_| b"Internal error: unable to create gas handler".to_vec())?;
+
+    let dispatch = dispatch.into_stored();
+
+    QueueOf::<T>::remove_all();
+
+    QueueOf::<T>::queue(dispatch).map_err(|_| b"Messages storage corrupted".to_vec())?;
+
+    let block_info = BlockInfo {
+        height: <frame_system::Pallet<T>>::block_number().unique_saturated_into(),
+        timestamp: <pallet_timestamp::Pallet<T>>::get().unique_saturated_into(),
+    };
+
+    let existential_deposit = <T as Config>::Currency::minimum_balance().unique_saturated_into();
+
+    let mut burned = 0;
+
+    while let Some(queued_dispatch) =
+        QueueOf::<T>::dequeue().map_err(|_| b"MQ storage corrupted".to_vec())?
+    {
+        let actor_id = queued_dispatch.destination();
+
+        let lazy_pages_enabled =
+            cfg!(feature = "lazy-pages") && lazy_pages::try_to_enable_lazy_pages();
+
+        let actor = ext_manager
+            .get_executable_actor(actor_id.into_origin(), !lazy_pages_enabled)
+            .ok_or_else(|| b"Program not found in the storage".to_vec())?;
+
+        let allocations_config = AllocationsConfig {
+            max_pages: gear_core::memory::WasmPageNumber(schedule.limits.memory_pages),
+            init_cost: schedule.memory_weights.initial_cost,
+            alloc_cost: schedule.memory_weights.allocation_cost,
+            mem_grow_cost: schedule.memory_weights.grow_cost,
+            load_page_cost: schedule.memory_weights.load_cost,
+        };
+
+        let gas_limit = T::GasHandler::get_limit(queued_dispatch.id().into_origin())
+            .ok()
+            .flatten()
+            .ok_or_else(|| b"Internal error: unable to get gas limit after execution".to_vec())?;
+
+        let journal = if lazy_pages_enabled {
+            core_processor::process::<LazyPagesExt, SandboxEnvironment<_>>(
+                Some(actor),
+                queued_dispatch.into_incoming(gas_limit),
+                block_info,
+                allocations_config,
+                existential_deposit,
+                ProgramId::from_origin(source),
+                actor_id,
+                u64::MAX,
+                T::OutgoingLimit::get(),
+                schedule.host_fn_weights.clone().into_core(),
+            )
+        } else {
+            core_processor::process::<Ext, SandboxEnvironment<_>>(
+                Some(actor),
+                queued_dispatch.into_incoming(gas_limit),
+                block_info,
+                allocations_config,
+                existential_deposit,
+                ProgramId::from_origin(source),
+                actor_id,
+                u64::MAX,
+                T::OutgoingLimit::get(),
+                schedule.host_fn_weights.clone().into_core(),
+            )
+        };
+
+        // TODO: Check whether we charge gas fee for submitting code after #646
+        for note in journal.iter() {
```

Comment:
I'd prefer to not call `.iter()` explicitly.
```suggestion
        for note in &journal {
```